### DISCUSSION TITLE
Querying the same data from the database for the same query present in multiple query fields

### DISCUSSION THREAD

**Discussion by SanthoshS20** [grafana/grafana#53714] [Observability]
**Problem/Idea:**
In the Loki query dashboard, same query present in multiple query fields result in fetching the same data multiple times from the database. Suppose if we have same query in more then 1 query fields, deleting/disabling/enabling anyone of the query result in API request and fetching the same data from the database.

**Impact:**
This impacts additional API resource utilization, time took to fetch it from the database. 

**Steps:**
1. Go to the Loki query dashboard.
2. Add a same query in the two query fields and run it.
3. API request will be sent with two separate query (Fetches the same data from the database for the same query).
4. Delete/Disable a query field, an API request will send and fetches the data which was already shown in the dashboard.

**Image:**
![image](https://user-images.githubusercontent.com/37366893/184530677-62216845-3c7d-49b2-ab9a-f6256fe3e444.png)

**Comment by torkelo** [comment]
why would you add two identical queries? In the 10 years of Grafana I have never seen a report that duplicate queries is an issue :) Do you see this a lot in the dashboard you use?

**Reply by SanthoshS20** [reply]
Yes @torkelo. Suppose if a user uses duplicate query button to create another query field, the run query button will be auto triggered if a user mentions to reload the page after every 1 min. I see this feature is to optimized instead of fetching the same data again from the database. Another case, consider a user duplicated the same query and the user didn't ran the run query button. After few seconds he didn't want that query field, he deleted/disabled. This results in sending API request to fetch the resources which was already present. I'm looking for an optimization to reduce the API requests.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
